MapboxNavigation/RouteMapViewController.swift

```diff
@@ -282,6 +308,25 @@ class RouteMapViewController: UIViewController {
     func mapView(_ mapView: MGLMapView, viewFor annotation: MGLAnnotation) -> MGLAnnotationView? {
         return navigationMapView(mapView, viewFor: annotation)
     }
+    
+    var cameraCurrentForRouteProgress: MGLMapCamera {
+        
+        guard !isInOverviewMode else {
+            return genericTiltedCamera
+        }
+        
+        guard let _ = routeController.routeProgress.currentLegProgress.upComingStep else {
+            return genericTiltedCamera
+        }
+        
+        if let _ = routeController.routeProgress.currentLegProgress.currentStep.intersections?.first?.outletRoadClasses?.contains(.motorway) {
```

Comment:
Oof. I see what you're doing here, but that's a doozy of a line. In this case, I might be tempted to use a let to define _what_ exactly i'm looking for (some semantic clue to the developer that's more informative then the call chain), and then use _chaining_ to do the test. i.e.:

```swift
if let currentIntersection = routeController.routeProgress.currentLegProgress.currentStep.intersections?.first,
   let outletTypes = currentIntersection.outletRoadClasses,
   outletTypes.contains(.motorway) {
```

IMHO this reads quite a bit more naturally, but YMMV.